thinc/backends/_custom_kernels.py

```diff
@@ -363,3 +409,56 @@ def hash(ids, seed, out=None, threads_per_block=128, num_blocks=128):
         (out, ids, out_size, in_size, ids.shape[0], seed),
     )
     return out
+
+
+def _check_compatible(out, shape: Optional[Tuple] = None):
+    assert out.dtype == "float32", "CUDA kernel can only handle float32"
+    if shape is not None and out.shape != shape:
+        msg = f"array has incorrect shape, expected: {shape}, was: {out.shape}"
+        raise ValueError(msg)
```

Comment:
It's not a big deal because it's an internal function, but I'd make the name a bit more explicit, maybe something like `_check_cuda_compatible` or `_check_float32_compat` or such. Because it comes across as a bit weird otherwise when this is called with just one argument (-> compatible to _what_?)

Reply:
I think this needs be renamed later anyway, since the `float32` check is replaced in the next PR which makes the kernels generic (also supporting `float64`).

Reply:
Right! Ok sure, then we can leave as is for now.